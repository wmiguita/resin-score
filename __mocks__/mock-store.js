import configureStore from "redux-mock-store"

const mockStore = configureStore([])

//TODO: find better way to keep reducers up to date and check for any test side effects
export const store = mockStore({
  atheletes: {
  },
  userInterface: {
    drawerOpen: false,
    selectedTab: 0
  }
})
