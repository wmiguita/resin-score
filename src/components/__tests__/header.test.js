import React from "react"
import { Provider } from "react-redux"
import renderer from "react-test-renderer"
import { store } from "mock-store"

import Header from "../header"

describe( "Header", () => {
  it( "should render Header", () => {
    // console.log( 'headerTest', gatsby )
    const header = renderer.create( <Provider store={ store }><Header /></Provider> ).toJSON()

    expect( header ).toMatchSnapshot()
  })
})

