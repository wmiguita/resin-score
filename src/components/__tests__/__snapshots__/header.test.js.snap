// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <button
      aria-label="open drawer"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
    </button>
    <h1
      className="MuiTypography-root MuiTypography-h6"
    >
      Atheletes
    </h1>
    <div
      className="MuiTabs-root"
    >
      <div
        className="MuiTabs-scroller MuiTabs-fixed"
        onScroll={[Function]}
        style={
          Object {
            "marginBottom": null,
            "overflow": "hidden",
          }
        }
      >
        <div
          aria-label="Athelete menu"
          className="MuiTabs-flexContainer"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-controls="list-atheletes"
            aria-selected={true}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
            disabled={false}
            id="listing"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Listing
            </span>
            <span
              className="PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 MuiTabs-indicator"
              style={Object {}}
            />
          </button>
          <button
            aria-controls="manage-athelete"
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
            disabled={false}
            id="manage"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper"
            >
              Manage
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</header>
`;
